tests/integration/query.rs

```diff
@@ -64,8 +64,8 @@ pub fn test_query_with_wildcard() {
             // stored inside the label, which is the one used for reporting anyway.
             // assert_eq!(contract1.types, contract2.types);
             assert_eq!(
-                contract1.label.types.as_ref(),
-                contract2.label.types.as_ref()
+                contract1.label.types.as_ref().clone().without_pos(),
```

Comment:
Could we rewrite `as_ref().clone()` to just `clone()`, or even get rid of it altogether? I can't see the types here, so I'm not sure, but it sounds fishy to take a ref and immediately clone it.

Reply:
but `clone()` takes by ref, so it doesn't move, does it?

Reply:
Ah, the actual answer is that it is an `Rc`, so if you clone directly, you're just cloning the `Rc`, not the inner value. Ideally we would use [`unwrap_or_clone`](https://doc.rust-lang.org/std/rc/struct.Rc.html#method.unwrap_or_clone), I've needed that for a long time, but it's still nightly, so for now it's probably the only solution.